src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -770,6 +846,29 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!--
+    ============================================================
+                                        BundlePublishDirectory
+
+    Bundle the items in PreBundlePublishDir into one file in PublishDir
+    ============================================================
+    -->
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Build.Bundle" />
+
+  <Target Name="BundlePublishDirectory" 
+          Condition="'$(PublishSingleFile)' == 'true'"
+          Inputs="$(PreBundlePublishDir)\**\*.*"
+          Outputs="$(PublishDir)\$(AssemblyName)$(_NativeExecutableExtension)">
+    <PropertyGroup>
+      <BundleArgs>--source $(PreBundlePublishDir) </BundleArgs>
```

Comment:
Try this with spaces in the path to the project file and spaces in the project file name / assembly file itself.